app/services/data_exports.rb

```diff
@@ -0,0 +1,293 @@
+require 'fileutils'
+require 'securerandom'
+require 'csv'
+
+module DataExports
+  class NoStoredFilesFoundError < StandardError; end
+
+  class DataStorage
+    # Public: downloads all transcription files for a given project,
+    # workflow, group or transcription
+    def resource_files_zip(resource)
+      # to do: include user id in top level directory name
+      directory_path = File.expand_path("~/data_exports_temp/downloaded_files/user_id_#{SecureRandom.uuid}")
+      FileUtils.mkdir_p(directory_path)
+
+      if resource.is_a?(Transcription)
+        if resource.files.attached?
+          transcription_folder = download_transcription_files(resource, directory_path)
+          return zip_files(directory_path, transcription_folder)
+        else
+          raise NoStoredFilesFoundError.new("No stored files found for transcription with id '#{resource.id}'")
+        end
+
+      # Transcription Group will come in as array of transcriptions
+      elsif resource.respond_to?('each')
+        zip_group(resource, directory_path)
+
+      elsif resource.is_a?(Workflow)
+        return zip_workflow(resource, directory_path)
+
+      elsif resource.is_a?(Project)
+        zip_project(resource, directory_path)
+      end
+    end
+
+    private
+
+    # download transcription files for a given transcription from storage to disk
+    # @param transcription [Transcription]: the transcription we want to retrieve files for
+    # @param directory_path [String]: path within which we will create the transcription file folder
+    # returns location of generated transcription folder
+    def download_transcription_files(transcription, directory_path)
+      transcription_folder = File.join(directory_path, "transcription_#{transcription.id}")
```

Comment:
we should really be using ruby tmp files to handle this use case instead of directories / files on the underlying FS to avoid artifacts / left over data being an issue with behaviour and/or disk space in the container.

I'd be happy to help pair on this if help is needed to switch to the tmp file solution.